subprojects/scala/src/main/java/org/gradle/api/plugins/scala/ScalaBasePlugin.java

```diff
@@ -122,7 +122,10 @@ private void configureConfigurations(final Project project, final Usage incremen
 
         Configuration zinc = project.getConfigurations().create(ZINC_CONFIGURATION_NAME);
         zinc.setVisible(false);
+        zinc.setCanBeConsumed(false);
```

Comment:
This is technically a breaking change. We could instead first deprecate this for consumption and delay attributes configuration for 8.0. Thoughts?

Reply:
I would think this "breakage" is ok, as in practice this should not be consumed by anyone. And if, it's easy to work around. (e.g. by just setting it back to `true` in your build script.)
We can add a sentence to the "potentially breaking changes" section of the upgrading guide.

Reply:
I reverted the changes to ScalaBasePlugin, leaving only non-breaking ones in this PR.
ScalaBasePlugin's zinc configuration I will handle in a separate change where I will deprecate it for consumption first and will then add required attributes together with a capability-workaround.